### PR TITLE
Validate Maven build caching samples

### PR DESCRIPTION
**Pull request by jprinet**
Here is a GH workflow to validate our Maven build caching samples.

The current version is good enough to catch unknown fields in a Mojo.
An unhandled field is only logged as a warning in the console when a GE backend is effectively plugged. This is not supported at the moment.

Let me know @bigdaz if this is what you were thinking about.

### CODE REVIEW

File: .github/workflows/build-caching-maven-samples.yml
Comment (bigdaz) [inline .github/workflows/build-caching-maven-samples.yml:11]: Where are these variables consumed? What impact do they have?

File: .github/workflows/build-caching-maven-samples.yml
Comment (bigdaz) [inline .github/workflows/build-caching-maven-samples.yml:29]: How does this verify that build caching is correctly configured?

**Review by bigdaz — CHANGES_REQUESTED** [review]
@jprinet I don't really understand what this workflow will check for, and how it works?
My preference would be to run the build connected to a real GE instance (so that local caching can be enabled).
The workflow would then effectively run Experiment 1 and verify the expected number of goals are FROM-CACHE.

**Comment by jprinet** [comment]
> @jprinet I don't really understand what this workflow will check for, and how it works? My preference would be to run the build connected to a real GE instance (so that local caching can be enabled). The workflow would then effectively run Experiment 1 and verify the expected number of goals are FROM-CACHE.

This actual workflow raises a failure if an unknown property is set in the plugin customization (in addition to any global failure due to syntax for instance). This can arise with a new version.
I do agree that a great improvement would be to connect to GE to get unhandled property and check caching.

File: .github/workflows/build-caching-maven-samples.yml
Comment (jprinet) [reply .github/workflows/build-caching-maven-samples.yml:11]: just a wrong copy/paste, deleting that

File: .github/workflows/build-caching-maven-samples.yml
Comment (jprinet) [reply .github/workflows/build-caching-maven-samples.yml:29]: it does not, build cache is disabled if no GE instance is connected

**Comment by bigdaz** [comment]
> This actual workflow raises a failure if an unknown property is set in the plugin customization (in addition to any global failure due to syntax for instance). This can arise with a new version.

OK got it. In my experience the most common issue is that the new Mojo version will have additional parameters, causing caching to stop working without any build failure. I'm OK with merging this PR as a first step, but I don't think it will catch the majority of issues that come with upgrading dependency versions.

**Comment by jprinet** [comment]
my impression is that added value is not much while we don't have a way to get the caching feature enabled.
I would keep that on hold before having a more complete solution.
wdyt @etiennestuder ?

**Comment by bigdaz** [comment]
@jprinet Do you know if e.grdev.net has Maven functionality enabled? If so, it would be nice to see if it could be used as a backing GE server for this repo. If it's possible, then we'd simply need to get approval to use e.grdev.net for this public repo: I'm not sure how important it is not to advertise that URL.

**Comment by runningcode** [comment]
To summarize:
My understanding is that this will test for a property being configured in the Gradle Enterprise configuration which does not actually exist in the Mojo.
This will not test for a new property being added in - for example - an update unless we connect it to a GE instance that has Maven caching enabled.

While the added functionality is not much, it does allow us to expand this in the future and connect it to a GE instance for proper testing.

After discussing with @etiennestuder we came to this conclusion: let's remove the functionality to connect the build to a GE instance with secrets and only merge the parts that check for an unknown property. Once that is merged we can circle back on adding the rest of the functionality.

File: .github/workflows/build-caching-maven-samples.yml
Comment (runningcode) [reply .github/workflows/build-caching-maven-samples.yml:26]: can we remove the `-Dorg.slf4j.simpleLogger.log.gradle.goal.cache=debug` line?

File: .github/workflows/build-caching-maven-samples.yml
Comment (runningcode) [inline .github/workflows/build-caching-maven-samples.yml:22]: why do we need yarn installed?

**Comment by jprinet** [comment]
Reverted to the version without requirement for a GE backend, please confirm that this is good to go @runningcode

File: .github/workflows/build-caching-maven-samples.yml
Comment (jprinet) [reply .github/workflows/build-caching-maven-samples.yml:22]: `exec-yarn-project` requires it

Comment (runningcode) [reply .github/workflows/build-caching-maven-samples.yml:22]: ah, of course!

**Review by runningcode — APPROVED** [review]
Looks great! Feel free to merge.

**Pull request merged**
